### ISSUE TITLE
nodegit fails to build for Electron on Windows

### ISSUE THREAD

**Issue by taylorzane** [nodegit/nodegit#738]
When I attempt to run `npm install` to build nodegit for electron, I am confronted with an error.

```
[nodegit] Everything is ready to go, attempting compilation
[nodegit] Building native node module.
{ [Error: Command failed: C:\Windows\system32\cmd.exe /s /c "C:\Users\me\git
\some-project\node_modules\nodegit\node_modules\.bin\node-gyp rebuild --target=0.3
3.4  --dist-url=https://gh-contractor-zcbenz.s3.amazonaws.com/atom-shell/dist"
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\12.0\bin\msbuild.exe` fail
ed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\me\git\some-project\node_m
odules\nodegit\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_proces
s.js:200:12)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\me\\git\\
some-project\\node_modules\\nodegit\\node_modules\\node-gyp\\bin\\node-gyp.js" "re
build" "--target=0.33.4" "--dist-url=https://gh-contractor-zcbenz.s3.amazonaws.c
om/atom-shell/dist"
gyp ERR! cwd C:\Users\me\git\some-project\node_modules\nodegit
gyp ERR! node -v v4.1.1
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
]
  killed: false,
  code: 1,
  signal: null,
  cmd: 'C:\\Windows\\system32\\cmd.exe /s /c "C:\\Users\\me\\git\\some-project
\\node_modules\\nodegit\\node_modules\\.bin\\node-gyp rebuild --target=0.33.4  -
-dist-url=https://gh-contractor-zcbenz.s3.amazonaws.com/atom-shell/dist"' }
undefined
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\
node_modules\\npm\\bin\\npm-cli.js" "install" "nodegit/nodegit#master"
npm ERR! node v4.1.1
npm ERR! npm  v2.14.4
npm ERR! code ELIFECYCLE

npm ERR! nodegit@0.5.0 install: `node lifecycleScripts/install`
npm ERR! Exit status 13
npm ERR!
npm ERR! Failed at the nodegit@0.5.0 install script 'node lifecycleScripts/insta
ll'.
npm ERR! This is most likely a problem with the nodegit package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node lifecycleScripts/install
npm ERR! You can get their info via:
npm ERR!     npm owner ls nodegit
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\me\git\some-project\npm-debug.log
```

These are the specs of my build environment (let me know if anything else is needed):
- Windows 7 SP1 x64
- Node v4.1.1
- Electron v0.33.4
- nodegit v0.5.0 (also tried a couple v0.4.x versions)
- Visual Studio 2013

For what it's worth, this exact repo builds just fine on my Mac running OS X 10.10.

**Comment by ssreekanth** [comment]
I have the same issue with my build environment (same as mentioned above in the bug report).

I had reported it a while ago [here](https://github.com/nodegit/nodegit/pull/702#issuecomment-141730234) and [here](https://github.com/nodegit/nodegit/pull/702#issuecomment-141751608), but so far no luck with the latest release 0.5.0.

@saper Hope we will have the fix soon.

**Comment by johnhaley81** [comment]
Now that https://github.com/mapbox/node-pre-gyp/pull/175 has landed I'm going to try and get out pre-builts for electron to help stop this insanity.

**Comment by tbranyen** [comment]
Woah

**Comment by johnhaley81** [comment]
Right?!

**Comment by taylorzane** [comment]
Fantastic! This is great news.

**Comment by taylorzane** [comment]
@johnhaley81 @tbranyen Any update on when this issue will be resolved?

**Comment by johnhaley81** [comment]
#783 is the next step :)

**Comment by joshball** [comment]
Hey @johnhaley81, so is https://github.com/nodegit/nodegit/pull/783 resolved now? It looks like some code was merged by you and maybe @joshaber?

**Comment by maxkorp** [comment]
We just published 0.7.0 a few moments ago, and has a lot of build fixes, so I would try that. I'd assume it's resolved (a number of windows specific issues were resolved).

**Comment by johnhaley81** [comment]
So going forward we are providing pre-built binaries for electron that you can download on all supported platforms (osx, linux, win32-64) for the current and the previous 2 minor versions of electron (currently 0.36.x, 0.35.x, and 0.34.x).

You shouldn't need to build nodegit anymore but can always force it to build for electron by running `node lifecycleScripts/install.js --electronVersion=v0.36.4` or whatever version of electron you want to target.

If you're still having issues we can reopen this but since we landed in atom/atom I know that it's compiling for electron across all platforms.
